396 display hal status in publication list

yanisvieilly opened the pull request.

yanisvieilly commented on src/app/[lang]/documents/page.tsx:
```diff
@@ -316,8 +324,50 @@ export default function DocumentsPage() {
+            halSubmitTypeToHalSubmitTypeIcon(halSubmitType)
+
+          // TODO: Implement collection logic
+          const isInCollection = halCollectionCodes.includes('test')
```
@jdp1ps The issue states "If hal_collection_codes contains the collection code of the laboratory of the researcher, Badge "Incollection"".

Is this the collection code of the laboratory of the current researcher (the current user)?

yanisvieilly replied:
The issue states "If hal_collection_codes contains the collection code of the laboratory of the researcher, Badge "Incollection""

Is this the laboratory of the logged in user, or the viewed as/"perspective" one?

yanisvieilly commented:
@jdp1ps I'll add a few tests tomorrow.

jdp1ps commented on src/app/[lang]/documents/page.tsx:
```diff
@@ -316,8 +328,86 @@ export default function DocumentsPage() {
+          const halSubmitTypeIcon =
+            halSubmitTypeToHalSubmitTypeIcon(halSubmitType)
+
+          const isResearchStructureInCollectionCodes = (
```
This pure function is related to data-logic, it could be moved :
- in the file, outside DocumentsPage component
- in a dedicated file as helper function
- in the Document class : 
`hasResearchStructureInCollectionCodes(perspective: IAgent | null): boolean`

yanisvieilly replied:
Agree, I ended up moving it to the `DocumentRecord` class.

yanisvieilly commented on src/app/lib/daos/PersonDAO.ts:
@jdp1ps `memberships` relation has been added to the Person slice

yanisvieilly commented on src/app/lib/daos/UserDAO.test.ts:
@jdp1ps `memberships` relation has been added to the User store slice

yanisvieilly commented on src/app/lib/daos/PersonDAO.ts:
```diff
@@ -333,6 +333,19 @@ export class PersonDAO extends AbstractDAO {
+        memberships: {
+          select: {
+            startDate: true,
+            endDate: true,
```
@jdp1ps Note that I'm not checking for the `startDate` and `endDate` of a membership at the moment (when displaying the Hal status badge). They are always `null` in the database. We might need to support this later?

jdp1ps replied:
the ldap data source used up to now does not provide this information. We may have to change this behavior for CSV or other sources (as you say, later)

yanisvieilly commented:
@jdp1ps I've added some tests, translations, filtered some of the membership fields added to the store, and fixed the Hal status chip style when it's on multiple lines. It's ready to be reviewed!
